src/SecTester.Scan/CI/VendorMatcher.cs

```diff
@@ -0,0 +1,96 @@
+using System;
+using System.Collections;
+using System.Linq;
+using System.Text.Json;
+
+namespace SecTester.Scan.CI;
+
+internal class VendorMatcher
+{
+  private readonly IDictionary _environment;
+
+  public VendorMatcher(IDictionary environment)
+  {
+    _environment = environment;
+  }
+
+  public bool MatchEnv(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => HasValue(element),
+      JsonValueKind.Object => MatchObject(element),
+      JsonValueKind.Array => element.EnumerateArray().All(HasValue),
+      _ => false
+    };
+  }
+
+  public bool MatchPr(JsonElement element)
+  {
+    return element.ValueKind switch
+    {
+      JsonValueKind.String => HasValue(element),
+      JsonValueKind.Object => NotEqual(element) || MatchObject(element),
+      _ => false
+    };
+  }
+
+  private bool MatchObject(JsonElement element)
+  {
+    return Any(element) || Includes(element) || HasOwnProperties(element);
+  }
+
+  private bool HasValue(JsonElement element)
```

Comment:
`Contains` just verifies if there is a property, but cannot say anything about value https://github.com/NeuraLegion/sectester-net/pull/94#discussion_r1045195231

```suggestion
  private bool HasProperty(JsonElement element)
```